dev-packages/cloudflare-integration-tests/runner.ts

```diff
@@ -0,0 +1,237 @@
+import type { Envelope, EnvelopeItemType } from '@sentry/core';
+import { normalize } from '@sentry/core';
+import { createBasicSentryServer } from '@sentry-internal/test-utils';
+import { spawn } from 'child_process';
+import { existsSync } from 'fs';
+import { join } from 'path';
+import { inspect } from 'util';
+import { expect } from 'vitest';
+
+const CLEANUP_STEPS = new Set<() => void>();
+
+export function cleanupChildProcesses(): void {
+  for (const step of CLEANUP_STEPS) {
+    step();
+  }
+  CLEANUP_STEPS.clear();
+}
+
+process.on('exit', cleanupChildProcesses);
+
+function deferredPromise<T = void>(
+  done?: () => void,
+): { resolve: (val: T) => void; reject: (reason?: unknown) => void; promise: Promise<T> } {
+  let resolve;
+  let reject;
+  const promise = new Promise<T>((res, rej) => {
+    resolve = (val: T) => {
+      done?.();
+      res(val);
+    };
+    reject = (reason: Error) => {
+      done?.();
+      rej(reason);
+    };
+  });
+  if (!resolve || !reject) {
+    throw new Error('Failed to create deferred promise');
+  }
+  return {
+    resolve,
+    reject,
+    promise,
```

Comment:
```suggestion
    resolve = (val: T) => {
      res(val);
    };
    reject = (reason: Error) => {
      rej(reason);
    };
  });
  if (!resolve || !reject) {
    throw new Error('Failed to create deferred promise');
  }
  return {
    resolve,
    reject,
    promise: promise.finally(() => done?.()),
```
If we could use node >= 22 it might be even more native `Promise.withResolver()`